Assets/Scripts/InteractionController.cs

```diff
@@ -158,13 +186,19 @@ public void ShowEarthMarkerInteraction(Vector3 pos, Quaternion rot, Color player
             }
             string interactionInfo = "Earth interaction at: " + latLng.ToString();
             Debug.Log(interactionInfo);
-            SimulationEvents.GetInstance().LocationChanged.Invoke(latLng, "Custom: ");
+            SimulationEvents.GetInstance().LocationChanged.Invoke(latLng, SimulationConstants.CUSTOM_LOCATION);
             CCLogger.Log(CCLogger.EVENT_ADD_INTERACTION, interactionInfo);
         }
     }
     public void SetEarthLocationPin(Vector3 pos, Quaternion rot, Color playerColor, bool isLocal)
     {
-        Vector2 latLng = Vector2.zero; // unset value
+        LatLng latLng = getEarthRelativeLatLng(pos);
+        AddOrUpdatePin(latLng, playerColor, manager.LocalUsername, manager.CurrentSimulationTime, true, true);
```

Comment:
Is `manager.LocalUsername` correct? If so, it seems like we could remove `playerColor` from the argument list, and also `isLocal` argument.

Reply:
There's definitely some tangled logic here - I'll attempt to resolve along with the datetime issues